airbyte-db/db-lib/src/main/java/io/airbyte/db/factory/DataSourceFactory.java

```diff
@@ -196,19 +198,23 @@ private DataSourceBuilder() {}
      *
      * @param connectionProperties custom jdbc_url_parameters containing information on connection
      *        properties
+     * @param driverClassName name of the JDBC driver
      * @return DataSourceBuilder class used to create dynamic fields for DataSource
      */
-    private static long getConnectionTimeoutMs(final Map<String, String> connectionProperties) {
-      final Duration connectionTimeout;
-      // TODO: the usage of CONNECT_TIMEOUT_KEY is Postgres specific, may need to extend for other
-      // databases
-      connectionTimeout =
-          connectionProperties.containsKey(CONNECT_TIMEOUT_KEY) ? Duration.ofSeconds(Long.parseLong(connectionProperties.get(CONNECT_TIMEOUT_KEY)))
-              : CONNECT_TIMEOUT_DEFAULT;
-      if (connectionTimeout.getSeconds() == 0) {
-        return connectionTimeout.toMillis();
+    private static long getConnectionTimeoutMs(final Map<String, String> connectionProperties, String driverClassName) {
```

Comment:
Can you add a test for this new functionality in `DataSourceFactoryTest.java`?

Reply:
It appears these changes were caught within the [build failures](https://github.com/airbytehq/airbyte/actions/runs/3444138664/jobs/5746451880#step:13:9164)

To add onto what Akash said, since this is a postgres specific feature, I would add a test primarily for postgres and keep a generic one to account for non-postgres behavior.

Reply:
> It appears these changes were caught within the [build failures](https://github.com/airbytehq/airbyte/actions/runs/3444138664/jobs/5746451880#step:13:9164)
> 
> To add onto what Akash said, since this is a postgres specific feature I would add a test primarily for postgres and keep a generic one to account for non-postgres behavior.

@akashkulk @ryankfu added tests for Postgres and MySQL to see the difference